internal/provisioner/providers.go

```diff
@@ -0,0 +1,14 @@
+package provisioner
+
+import "fmt"
+
+// providerAWS is the cloud provider AWS.
+const providerAWS = "aws"
+
+func checkProvider(provider string) error {
```

Comment:
Might we consider a signature of `checkProvider(provider string) (string, error)`, returning the valid provider and allowing us to internalize the `strings.ToLower` in this method, and even support other future multi-mappings (e.g. `"amazon"` or whatever).